pkg/queue/cluster_queue_best_effort_fifo_test.go

```diff
@@ -20,160 +20,44 @@ import (
-			admissibleWorkloadsToRequeue:   []*workload.Info{workload.NewInfo(workloads[1])},
-			wantActiveWorkloads:            sets.NewString(workloads[0].Name, workloads[1].Name),
-			wantPending:                    2,
+		RequeueReasonNamespaceMismatch: {
+			wantInadmissible: true,
 		},
-		"re-queue inadmissible workload and flush": {
-			workloadsToAdd:                 []*kueue.Workload{workloads[0]},
-			inadmissibleWorkloadsToRequeue: []*workload.Info{workload.NewInfo(workloads[1])},
-			queueInadmissibleWorkloads:     true,
-			wantActiveWorkloads:            sets.NewString(workloads[0].Name, workloads[1].Name),
-			wantPending:                    2,
-		},
-		"update inadmissible workload": {
-			workloadsToAdd:                 []*kueue.Workload{workloads[0]},
-			inadmissibleWorkloadsToRequeue: []*workload.Info{workload.NewInfo(workloads[1])},
-			workloadsToUpdate:              []*kueue.Workload{updatedWorkloads[1]},
-			wantActiveWorkloads:            sets.NewString(workloads[0].Name, workloads[1].Name),
-			wantPending:                    2,
-		},
-		"delete inadmissible workload": {
-			workloadsToAdd:                 []*kueue.Workload{workloads[0]},
-			inadmissibleWorkloadsToRequeue: []*workload.Info{workload.NewInfo(workloads[1])},
-			workloadsToDelete:              []*kueue.Workload{workloads[1]},
-			queueInadmissibleWorkloads:     true,
-			wantActiveWorkloads:            sets.NewString(workloads[0].Name),
-			wantPending:                    1,
+		RequeueReasonGeneric: {
+			wantInadmissible: true,
 		},
 	}
 
-	for name, test := range tests {
-		t.Run(name, func(t *testing.T) {
-			cq, err := newClusterQueueBestEffortFIFO(clusterQueue)
-			if err != nil {
-				t.Fatalf("Failed creating ClusterQueue %v", err)
-			}
-
-			for _, w := range test.workloadsToAdd {
-				cq.PushOrUpdate(workload.NewInfo(w))
-			}
-
-			for _, w := range test.inadmissibleWorkloadsToRequeue {
-				cq.RequeueIfNotPresent(w, false)
-			}
-			for _, w := range test.admissibleWorkloadsToRequeue {
-				cq.RequeueIfNotPresent(w, true)
-			}
-
-			for _, w := range test.workloadsToUpdate {
-				cq.PushOrUpdate(workload.NewInfo(w))
-			}
-
-			for _, w := range test.workloadsToDelete {
-				cq.Delete(w)
+	for reason, test := range tests {
+		t.Run(string(reason), func(t *testing.T) {
+			cq, _ := newClusterQueueBestEffortFIFO(&kueue.ClusterQueue{
+				Spec: kueue.ClusterQueueSpec{
+					QueueingStrategy: kueue.StrictFIFO,
+				},
+			})
+			wl := utiltesting.MakeWorkload("workload-1", defaultNamespace).Obj()
+			if ok := cq.RequeueIfNotPresent(workload.NewInfo(wl), reason); !ok {
+				t.Error("failed to requeue nonexistent workload")
 			}
 
-			if test.queueInadmissibleWorkloads {
-				cq.QueueInadmissibleWorkloads()
+			_, gotInadmissible := cq.(*ClusterQueueBestEffortFIFO).inadmissibleWorkloads[workload.Key(wl)]
+			if diff := cmp.Diff(test.wantInadmissible, gotInadmissible); diff != "" {
```

Comment:
nit: don't use Diff for booleans

Reply:
why not?

Reply:
the reason I sometimes like it is because it makes the error message look standard just like any other comparison we do.

Reply:
It makes the test output longer... But your argument makes sense too.